activation/poet.go

```diff
@@ -134,6 +142,14 @@ func NewHTTPPoetClient(server types.PoetServer, cfg PoetConfig, opts ...PoetClie
 		CheckRetry:   checkRetry,
 	}
 
+	submitChallengeClient := &retryablehttp.Client{
+		RetryMax:     math.MaxInt,
+		RetryWaitMin: cfg.RequestRetryDelay,
+		RetryWaitMax: 2 * cfg.RequestRetryDelay,
+		Backoff:      retryablehttp.LinearJitterBackoff,
+		CheckRetry:   checkRetry,
```

Comment:
NIT: when submitting a challenge, it doesn't make sense to retry on 404 (which `checkRetry` does). Consider using the default:
```suggestion
```